File: android/src/main/kotlin/com/bitmovin/player/flutter/FullscreenHandlerProxy.kt
package com.bitmovin.player.flutter

import com.bitmovin.player.api.ui.FullscreenHandler
import io.flutter.plugin.common.MethodChannel

class FullscreenHandlerProxy(
    override var isFullscreen: Boolean,
    private val methodChannel: MethodChannel,
) : FullscreenHandler {
    override fun onDestroy() {
        // no-op
    }

    override fun onFullscreenExitRequested() {
        isFullscreen = false
        runOnMainThread {
            methodChannel.invokeMethod(Methods.EXIT_FULLSCREEN, null)
        }
    }

    override fun onFullscreenRequested() {
        isFullscreen = true
        runOnMainThread {
            methodChannel.invokeMethod(Methods.ENTER_FULLSCREEN, null)
        }
    }

    override fun onPause() {
        // no-op
    }

    override fun onResume() {
        // no-op
    }
}

File: ios/bitmovin_player/Sources/bitmovin_player/FullscreenHandlerProxy.swift
import BitmovinPlayer
import Flutter
import Foundation

internal class FullscreenHandlerProxy: NSObject, FullscreenHandler {
    private(set) var isFullscreen: Bool
    private let methodChannel: FlutterMethodChannel

    init(
        isFullscreen: Bool,
        methodChannel: FlutterMethodChannel
    ) {
        self.isFullscreen = isFullscreen
        self.methodChannel = methodChannel
    }

    func onFullscreenRequested() {
        isFullscreen = true
        methodChannel.invokeMethod(Methods.enterFullscreen, arguments: nil)
    }

    func onFullscreenExitRequested() {
        isFullscreen = false
        methodChannel.invokeMethod(Methods.exitFullscreen, arguments: nil)
    }
}
